File: ejercicio 1/ejercicio1.py
def ternaPitagorica(a,b,c):
    return a**2 + b**2 == c**2

if ternaPitagorica(3,4,5):
    print("Es una terna pitagorica")
else:
    print("No es una terna pitagorica")
    
def ciclo_terna_pitagorica():
    for lado1 in range(1,501):
        for lado2 in range(1,501):
            for lado3 in range(1,501):
                if ternaPitagorica(lado1,lado2,lado3):
                    print(f"Terna pitagorica encontrada:  ({lado1}, {lado2}, {lado3})");
                    
ciclo_terna_pitagorica();

File: EJERCICIOS VARIOS/ejercicio 1/ejercicio1.cpp
#include <stdio.h>
bool ternaPitagorica(int a, int b, int c)
{
    return a * a + b * b == c * c;
}

int ciclo_terna_pitagorica()
{

    for (int lado1 = 1; lado1 <= 500; lado1++)
        for (int lado2 = 1; lado2 <= 500; lado2++)
            for (int lado3 = 1; lado3 <= 500; lado3++)
            {
                if (ternaPitagorica(lado1, lado2, lado3))
                    printf("Ternas pitoagoricas encontrads: (%d,%d,%d)\n", lado1, lado2, lado3);
            }
}

int main(void)
{
    //    if(ternaPitagorica(3,4,5)){
    //     printf("Es una terna pitagorica");
    //    }else{
    //     printf("No es una terna pitagorica");
    //    }
    ciclo_terna_pitagorica();
    return 0;
}
